File: 2024_java_basics/11-while-loops/tasks/05-account-balance.java
import java.util.Scanner;

public class SumNumbers {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        String input = scanner.nextLine();
        double total = 0.0;

        while (!input.equals("NoMoreMoney")) {
            try {
                double amount = Double.parseDouble(input);

                if (amount < 0) {
                    System.out.println("Invalid operation!");
                    break;
                }

                System.out.printf("Increase: %.2f%n", amount);
                total += amount;

                input = scanner.nextLine();
            } catch (NumberFormatException e) {
                System.out.println("Invalid input format!");
                input = scanner.nextLine();
            }
        }

        System.out.printf("Total: %.2f%n", total);

        scanner.close();
    }
}

File: 2022_c#_basics/11-while-loop/05-account-balance.cs
using System;

namespace _03_Sum_Numbers
{
    internal class Program
    {
        static void Main(string[] args)
        {
            string n = Console.ReadLine();
            double k = 0;

            while (n != "NoMoreMoney")
            {
                double n1 = double.Parse(n);

                if (n1 < 0)
                {
                    Console.WriteLine("Invalid operation!");
                    break;
                }

                Console.WriteLine($"Increase: {n1:f2}");
                k = k + n1;

                n = Console.ReadLine();
            }

            Console.WriteLine($"Total: {k:f2}");
        }
    }
}
